internal/uvm/plan9.go

```diff
@@ -12,10 +12,21 @@ import (
 	"github.com/Microsoft/hcsshim/osversion"
 )
 
+// Plan9Share is a struct containing host paths for the UVM
 type Plan9Share struct {
+	// UVM resource belongs to
+	VM            *UtilityVM
 	name, uvmPath string
 }
 
+// Release frees the resources of the corresponding Plan9 share
+func (p9 *Plan9Share) Release(ctx context.Context) error {
+	if err := p9.VM.RemovePlan9(ctx, p9); err != nil {
```

Comment:
You certainly can do this in steps but its a little odd to me to keep `RemovePlan9` (and others) as exported. This should really be the release function of the struct and we dont need a vm.Remove... function.

Reply:
I completely agree. All of the vm related functions that are called in the Release methods can be unexported now. Kind of slipped my brain

Reply:
The RemoveSCSI and RemoveVSMB are the ones blocking all of them from being unexported. They're called in places where currently there is no access to a vsmb or scsi mount and they're simply provided with a path and then look in the uvm's scsiLocations and vsmbFile/DirShares matrix/map to remove them. RemovePlan9 and RemovePipe can be unexported as of now as theyre only called from the uvm package/have access to a pipe or plan9 object. Alot would need to be refactored to make RemoveVSMB or RemoveSCSI unexported as of now though. Many tests simply loop through an arr of paths and call uvm.RemoveSCSI/VSMB with the paths so these would all need to be updated (which is doable obviously) I just havent looked into the tests side of things much yet to know if theres more pain points. I think for right now I will unexport Pipe and Plan9 and work out a plan for the others.

Reply:
I also think we should try to refactor this so that we can change the `Remove*` functions to be unexported. We should ensure all resource cleanup is done using the resource types that this change implements.

@dcantah can you make a list of the painful places to change? If it ends up being too much we can possibly do that as a separate PR.

Reply:
I feel its better suited for another PR for now. This would be grouped in with the layers cleanup that you mention below. Here's some pain points though, it's all SCSI and VSMB related. 
https://github.com/microsoft/hcsshim/blob/432e537acbeb74fed644b5b34a781b2675cae9ab/internal/hcsoci/layers.go#L73

https://github.com/microsoft/hcsshim/blob/432e537acbeb74fed644b5b34a781b2675cae9ab/internal/hcsoci/layers.go#L82

Tests like these where they're just based off paths would all have to be reworked. 
https://github.com/microsoft/hcsshim/blob/432e537acbeb74fed644b5b34a781b2675cae9ab/test/functional/uvm_scsi_test.go#L76